### PR TITLE
[NPU] remove ineffective params for ASCEND NPU

### PR DESCRIPTION
**Pull request by sunyi0505**
## Summary

remove ineffective params for ASCEND NPU and fix ut.

## Details

1. Refer to docs at: https://github.com/triton-lang/triton-ascend/blob/e9fcd7ddabbbc6de9579a18e507e8f12afbc840c/docs/zh/triton_api/Memory_Pointer_Ops/tl.store.md?plain=1#L31-L32 cache_modifier and eviction_policy are ineffective for Ascend devices. so, remove these params from ascend ops.

2. When running the full set of unit tests, certain test cases for the fused_add_rms_norm kernel fail, whereas standalone unit test execution for this kernel passes without issues. Further investigation shows that the full-test suite generates different random input values which hit an edge case: under bfloat16 with casting_mode="none", the numerical error of weight.grad sits right at the tolerance threshold. We fix the random seed to enforce identical random inputs.

## Testing Done

image

Tested with Atlas 800T A3(X86)

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by sunyi0505** [comment]
@Tcc0403 This pr is ready for review. Thanks!

File: test/transformers/test_fused_add_rms_norm.py [hunk 1]
```diff
@@ -124,6 +124,7 @@ def forward(self, x, residual):
     ],
 )
 def test_correctness(bs, sl, hd, dtype, atol, rtol, reference, offset, casting_mode, in_place):
+    set_seed(42)
```
Comment (Tcc0403) [inline test/transformers/test_fused_add_rms_norm.py:127]: why do we need to set seed here? is it a flaky test?

Comment (sunyi0505) [reply test/transformers/test_fused_add_rms_norm.py:127]: Yes — it is explicitly marked flaky:
@pytest.mark.flaky(reruns=3, reruns_delay=2)

Without resetting the random seed, previously executed parameterized test cases advance the global RNG state, causing generated input data to depend on execution order. This test consistently fails when I run the full UT suite on Ascend Atlas 800T A3.

**Review by Tcc0403 — APPROVED** [review]
LGTM

File: test/transformers/test_fused_add_rms_norm.py (hunk 1 above)
Comment (Tcc0403) [reply test/transformers/test_fused_add_rms_norm.py:127]: We should make it a pytest fixture to auto use `set_seed()` in conftest for reproducibility.

Comment (sunyi0505) [reply test/transformers/test_fused_add_rms_norm.py:127]: > We should make it a pytest fixture to auto use `set_seed()` in conftest for reproducibility.

Okay, I will fix it in the later PR.
